lib/vmpooler/pool_manager.rb

```diff
@@ -292,6 +292,100 @@ def destroy_vm(vm, pool)
       end
     end
 
+    def create_vm_snapshot(vm, snapshot_name)
+      Thread.new do
+        _create_vm_snapshot(vm, snapshot_name)
+      end
+    end
+
+    def _create_vm_snapshot(vm, snapshot_name)
```

Comment:
This method should do some scrutiny of `snapshot_name`. Or is that left to `CreateSnapshot_Task`?